### PR TITLE
task(all): Get new syncstorage-rs service running

### PR DESCRIPTION
**Pull request by dschom**
## Because

- Locally the sync service wasn't running.
- We couldn't test sync flows against a local sync server
- There's a new version of the sync server that uses rust

## This pull request

- Sets up a dockerfile for syncstorage-rs that is tailored to fxa's local stack
- Updates pm2 sync job to run this docker container and initialize its databases.
- Updates firefox config/profile to point at this service
- Starts firefox up with the FIREFOX_DEBUGGER=true env.
  - Useful for debugging sync
  - Useful for debugging FxA web channel messages

## Issue that this pull request solves

Closes: FXA-9669

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [ ] If applicable, I have modified or added tests which pass locally.
- [ ] I have added necessary documentation (if appropriate).
- [ ] I have verified that my changes render correctly in RTL (if appropriate).

## Other information (Optional)

The best way to test this is by running `yarn firefox` which setups a firefox instance that is configured to work with a local sync server. In order to test sync, try the following:

- Run `yarn start`
- Run the following query on your local db, and see that there are no entries. `SELECT * FROM syncstorage.bso;`
- Run `yarn firefox`. We will call this FF1
- Run `yarn firefox` again. We will call this FF2
- In FF1:
  - Go to sign in to sync
  - Sign in/up with an account
  - Go back to the hamburger menu and click sync
  - Run the following query in your local sqldb, and see that there are now entries, `SELECT * FROM syncstorage.bso;`
  - Add a bookmark
  - Sign into another account somewhere, an choose to save the password. 
    _(To test, I used `accounts.stage.mozaws.net`, and created a new account `$MYEMAIL+synctest001@mozilla.com`)_
  - Go to the hamburger menu, select passwords.
  - Go to the hamburger menu, and select sync now. _(This will ensure changes are sent to the sync storage.)_
- In FF2:
  - Go to sign in to sync
  - Sign in with the account used by FF1
  - Go back to the hamburger menu and click sync
  - Check the sync menu and see that the site you booked marked in FF is showing up.
  - Next, go to hamburger menu, and select passwords.
  - Depending on whether or not the sync has already happened, you should see your saved password from FF1. _(If you don't see them wait a second, and try clicking `sync now` again in FF2.)
  
From FF1:
image
image
image

From FF2
image
image

### CODE REVIEW

File: _dev/docker/sync/sync-for-fxa/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+WORKDIR /app
+
+# Add extra libs and setup python
+RUN apt-get update && \
+    apt-get install -y --no-install-recommends \
+    cmake golang-go python3-dev python3-pip python3-setuptools python3-wheel python3.11-venv libmariadb-dev-compat libmariadb-dev
+
+# Checkout the source, and switch to tag 0.17.2 which is what this was last tested with.
+RUN git clone https://github.com/mozilla-services/syncstorage-rs.git . && \
+    git checkout 0.17.2
```
Comment (chenba) [inline _dev/docker/sync/sync-for-fxa/Dockerfile:17]: `git clone --branch 0.17.2 --depth 1 https://github.com/mozilla-services/syncstorage-rs.git`?

**Comment by chenba** [comment]
It seems to work, kind of.  The second browser can see the sync'd tabs of the first browser.  But the first browser doesn't seem to know about the second browser at all.  I'm not sure it's even related to this patch.

**Review by chenba — APPROVED** [review]
🥇

**Review by vbudhram — COMMENTED** [review]
@dschom Thank you! I tried running this locally and had a conflict with port 5000 on osx, looks like `ControlCenter` app is using that port? Could be just my computer though.

**Comment by bbangert** [comment]
I presume this doesn't work in a dev container because Firefox can't be started up there. Are there alternate instructions on how to run Firefox correctly when using a dev container?

**Comment by chenba** [comment]
> I presume this doesn't work in a dev container because Firefox can't be started up there. Are there alternate instructions on how to run Firefox correctly when using a dev container?

I haven't tried but it should work since all the port numbers are the same when running with dev container.

**Comment by jonalmeida** [comment]
> @dschom Thank you! I tried running this locally and had a conflict with port 5000 on osx, looks like `ControlCenter` app is using that port? Could be just my computer though.

This is a known issue (undocumented) that we've had before and the setups were to disable controlcenter, it would be nice if we could avoid that from happening.

File: _dev/docker/sync/sync-for-fxa/Dockerfile (hunk 1 above)
Comment (dschom) [reply _dev/docker/sync/sync-for-fxa/Dockerfile:17]: Good call!

**Comment by dschom** [comment]
> > @dschom Thank you! I tried running this locally and had a conflict with port 5000 on osx, looks like `ControlCenter` app is using that port? Could be just my computer though.
> 
> +1. This was a known issue (undocumented) that we've had before and the setups were to disable controlcenter, it would be nice if we could avoid that from happening.

I'll change the port. Not a big deal. Thanks for point this out.

**Comment by dschom** [comment]
> > I presume this doesn't work in a dev container because Firefox can't be started up there. Are there alternate instructions on how to run Firefox correctly when using a dev container?
> 
> I haven't tried but it should work since all the port numbers are the same when running with dev container.

Yep, to be clear. You'd need to start firefox (eg `yarn firefox`) from outside the dev container.

**Pull request merged**